### PR TITLE
Change config file format from dict based to list based

### LINKED ISSUE #49: Change config format from dict to list of projects
In order to better and much more simply facilitate the hierarchy and clone order and more where we might want to clone a second repo into some already cloned folder it would make much more sense and make it simpler to process and parse repos in order if the user would make the repo ordering.

Example of new format would be

```
repos:
  - name: books
    url: git@.../books.git
    revision:
      branch: master
  - name: windows
    url: git@.../windows.git
    revision:
      branch: master
```

Things that is affected by this is plenty, here is a few things that probably needs to be rewritten

- Parsing of config file when creating Core class
- The use and extraction of data within all existing commands, instead of iterating over key:value pairs it needs to be iterate over a list and extract the name from elsewhere instead.
- When init a new config file the default config needs to generate this correct syntax
- A possible

A possible good idea within these kinds of changes is to make a more dedicated and separate migration section within documentation that describes how a user migrates between different versions of the syntax. Another idea is to implement "versions" of the syntax that can be used to validate or check if this version of the cli really works with or supports a specific version of the config file or not. This however might be another ticket and probably should not be done in here right now

### PR DESCRIPTION
**Pull request by Grokzen**
Fixed #49

### CODE REVIEW

File: subgit/core.py [hunk 1]
```diff
@@ -439,7 +451,8 @@ def pull(self, names):
                     # branch or where the origin HEAD is pointing to. After we clone we can then move our
                     # repo to the correct revision we want.
                     repo = Repo.clone_from(
-                        config["repos"][name]["url"],
+                        # config["repos"][name]["url"],
```
Comment (holmboe) [inline subgit/core.py:454]: Remove commented code?

**Review by holmboe — APPROVED** [review]
LGTM (1 minor comment)

File: subgit/core.py (hunk 1 above)
Comment (Grokzen) [inline subgit/core.py:454]: @holmboe Good catch, yes it should be removed

Comment (Grokzen) [reply subgit/core.py:454]:
```suggestion
```

**Pull request merged**